presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushDownTableConstraints.java

```diff
@@ -143,7 +145,7 @@ private boolean planChanged(PlanNode rewrittenPlan, FilterNode oldPlan, Lookup l
         }
 
         FilterNode rewrittenFilter = (FilterNode) rewrittenPlan;
-        if (!rewrittenFilter.getPredicate().equals(oldPlan.getPredicate())) {
+        if (!ImmutableSet.copyOf(extractConjuncts(rewrittenFilter.getPredicate())).equals(ImmutableSet.copyOf(extractConjuncts(oldPlan.getPredicate())))) {
```

Comment:
That's true.   would that happen as part of this rule?  I could use Expression equivalence instead